### PR TITLE
Update docs to use myst-parser (and be written in .md)

### PR DESCRIPTION
**Pull request by wlach**
Per discussion with @davidism, he seemed positive on moving the docs to MyST-flavored markdown.

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -16,23 +16,25 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-sys.path.insert(0, os.path.abspath(".."))
+sys.path.insert(0, os.path.abspath("../src"))
```
Comment (wlach) [inline docs/conf.py:19]: This is actually a bugfix.

Comment (davidism) [reply docs/conf.py:19]: This shouldn't be needed. `pip install -e .` in your local env (tox does it automatically) should already put the library on the path.

Comment (wlach) [reply docs/conf.py:19]: Ah ok, it was local docs builds in my virtualenv that weren't working. Do we have a best practice for generating previews (ideally live previews) in pallets projects? I didn't see one but maybe I wasn't looking hard enough

File: docs/conf.py [hunk 2]
```diff
@@ -10,29 +10,23 @@
...
 
 # Add any Sphinx extension module names here, as strings. They can be extensions
 # coming with Sphinx (named 'sphinx.ext.*') or your custom ones.
 extensions = [
+    "myst_parser",
     "sphinx.ext.autodoc",
     "sphinx.ext.intersphinx",
     "pallets_sphinx_themes",
 ]
+myst_enable_extensions = ["colon_fence"]
```
Comment (davidism) [inline docs/conf.py:23]: Let's be consistent and use triple backticks for fences in all places. It doesn't look like colon fences are needed to disambiguate anything.

Also see Flask-Orjson for a docs config that I used for myst: https://github.com/pallets-eco/flask-orjson/blob/main/docs/conf.py#L29-L32

File: docs/conf.py [hunk 3]
```diff
@@ -10,29 +10,23 @@
...
 
 # Add any paths that contain templates here, relative to this directory.
 templates_path = ["_templates"]
 
 # The suffix of source filenames.
-source_suffix = ".rst"
+source_suffix = [".md", ".rst"]
```
Comment (davidism) [inline docs/conf.py:29]: I think this line can be removed, it doesn't look like I needed it in Flask-Orjson.

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+The message can contain a body and/or HTML:
+
+```py
+msg.body = "testing"
+msg.html = "<b>testing</b>"
+```
+
+Finally, to send the message, you use the `Mail` instance configured with your Flask application:
+
+```
```
Comment (davidism) [inline docs/index.md:140]: Missing language mark.

File: docs/index.md [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+## Unit tests and suppressing emails
+
+When you are sending messages inside of unit tests, or in a development
+environment, it's useful to be able to suppress email sending.
+
+If the setting `TESTING` is set to `True`, emails will be
+suppressed. Calling `send()` on your messages will not result in
+any messages being actually sent.
+
+Alternatively outside a testing environment you can set `MAIL_SUPPRESS_SEND` to **True**. This
```
Comment (davidism) [inline docs/index.md:197]: Use code instead of bold formatting.

File: docs/conf.py (hunk 2 above)
Comment (wlach) [reply docs/conf.py:23]: Yeah, it's [rst2myst](https://github.com/executablebooks/rst-to-myst) which uses colon fence by default, I believe there's an option to make it uses backticks instead. Easy to fix after the fact ofc.

File: docs/index.md (hunk 4 above)
Comment (wlach) [reply docs/index.md:140]: Yeah this wasn't added by default and I put them in after the rst2myst conversion. Must have missed a few. I'll do another run through.

File: docs/index.md (hunk 5 above)
Comment (wlach) [reply docs/index.md:197]: I think this was the way it was originally written but happy to change it!

Comment (davidism) [reply docs/index.md:197]: Yeah, I figured, but this was the one inconsistent use so might as well fix it.

**Comment by wlach** [comment]
@davidism Ready for another look.

**Pull request merged**